src/main/java/com/product/dao/ProductDao.java

```diff
@@ -0,0 +1,20 @@
+package com.product.dao;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.RestTemplate;
+
+import com.product.model.Customer;
+import com.product.model.Product;
+
+@Repository
+public interface ProductDao extends MongoRepository<Product, Integer> {
+	
+	@Autowired
+	RestTemplate restTemplate = new RestTemplate();
```

Comment:
Why dont you make a `@Bean` of RestTemplate, and inject it by autowiring in all the classes. 

You dont have to initialize the RestTemplate with `new RestTemplate()` in every class that  way

Reply:
Done